sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -60,13 +60,25 @@ trait ReadStateStore {
   /** Version of the data in this store before committing updates. */
   def version: Long
 
+  /** Whether composite key is used for state store. */
+  var useCompositeKey: Boolean = false
```

Comment:
This is an ugly implementation. We may consider use similar ways as `useMultipleValuesPerKey`. But we may get lost in `TransformWithStateExec` in calling `createAndInit`?